src/high-level/scripts/papi_hl_output_writer.py

```diff
@@ -471,46 +498,72 @@ def write_json_file(data, file_name):
     outfile.write(to_unicode(str_))
     print (str_)
 
+def main(format, type, notation, source_dir = None, source_file = None):
+    if (format == "json"):
+        if source_dir != None:
+            json = merge_json_files(source_dir)
+        else:
+            json = parse_source_file(source_file)
 
-def main(source, format, type, notation):
-  if (format == "json"):
-    json = merge_json_files(source)
-
-    if type == 'detail':
-      if notation == 'derived':
-        write_json_file(derive_json_object(json), 'papi.json')
-      else:
-        write_json_file(json, 'papi.json')
-
-    #summarize data over regions with the same name, threads and ranks
-    if type == 'summary':
-      if notation == 'derived':
-        write_json_file(sum_json_object(json, True), 'papi_sum.json')
-      else:
-        write_json_file(sum_json_object(json), 'papi_sum.json')
+        if type == 'detail':
+            if notation == 'derived':
+                write_json_file(derive_json_object(json), 'papi.json')
+            else:
+                write_json_file(json, 'papi.json')
 
-  else:
-    print("Format not supported!")
+        #summarize data over regions with the same name, threads and ranks
+        if type == 'summary':
+            if notation == 'derived':
+                write_json_file(sum_json_object(json, True), 'papi_sum.json')
```

Comment:
Currently the output file is hardcoded to just be `papi_sum.json`, we possibly could add another flag that allows a user to pass an output filename.